src/clients/c++/experimental_api_v2/library/common.h

```diff
@@ -0,0 +1,561 @@
+  /// value is an empty string which means the server will select the
+  /// version of the model based on its internal policy.
+  std::string model_version_;
+  /// An identifier for the request. If specified will be returned
+  /// in the response. Default value is an empty string which means no
+  /// request_id will be used.
+  std::string request_id_;
+  /// The unique identifier for the sequence being represented by the
+  /// object. Default value is 0 which means that the request does not
+  /// belong to a sequence.
+  uint64_t sequence_id_;
+  /// Indicates whether the request being added marks the start of the
+  /// sequence. Default value is False. This argument is ignored if
+  /// 'sequence_id' is 0.
+  bool sequence_start_;
+  /// Indicates whether the request being added marks the end of the
+  /// sequence. Default value is False. This argument is ignored if
+  /// 'sequence_id' is 0.
+  bool sequence_end_;
+  /// Indicates the priority of the request. Priority value zero
+  /// indicates that the default priority level should be used
+  /// (i.e. same behavior as not specifying the priority parameter).
+  /// Lower value priorities indicate higher priority levels. Thus
+  /// the highest priority level is indicated by setting the parameter
+  /// to 1, the next highest is 2, etc. If not provided, the server
+  /// will handle the request using default setting for the model.
+  uint64_t priority_;
+  /// The timeout value for the request, in microseconds. If the request
+  /// cannot be completed within the time the server can take a
+  /// model-specific action such as terminating the request. If not
+  /// provided, the server will handle the request using default setting
+  /// for the model.
+  uint64_t timeout_;
+};
+
+//==============================================================================
+/// An interface for InferInput object to describe the model input for
+/// inference.
+///
+class InferInput {
+ public:
+  /// Create a InferInput instance that describes a model input.
+  /// \param infer_input Returns a new InferInput object.
+  /// \param name The name of input whose data will be described by this object.
+  /// \param dims The shape of the input.
+  /// \param datatype The datatype of the input.
+  /// \return Error object indicating success or failure.
+  static Error Create(
+      InferInput** infer_input, const std::string& name,
+      const std::vector<int64_t>& dims, const std::string& datatype);
+
+  /// Gets the name of input being described by this object
+  /// \return The name of the input
+  const std::string& Name() const { return name_; }
+
+  /// Gets the size of data set into this input in bytes.
+  /// \return The size of data in bytes.
+  int64_t ByteSize() const { return byte_size_; }
+
+  /// Gets the size of data which will be send on the wire.
+  /// \return The size of data to be sent on wire in bytes.
+  size_t TotalSendByteSize() const { return total_send_byte_size_; }
```

Comment:
Why is this needed and why is it different than ByteSize()?

Reply:
Removed. It was carry over from our current V1. It is different in sense total_send_byte_size_ will be 0 for shared_memory inputs. Will add later if the need arises.